addons/pos_self_order/models/product_product.py

```diff
@@ -101,3 +101,8 @@ def _send_availability_status(self):
                 config._notify('PRODUCT_CHANGED', {
                     'product.product': self.read(self._load_pos_self_data_fields(config.id), load=False)
                 })
+
+    def _find_record_check_access(self, access_token, field):
+        if field == "image_512":
+            return self.sudo()
+        return super()._find_record_check_access(access_token, field)
```

Comment:
`pos_self_order` depends on `pos_restaurant` which depends on `point_of_sale` so `image_128` is handled by `point_of_sale`.